File: localmart/order-service/src/main/java/com/localmart/order_service/event/OrderPlacedEvent.java
package com.localmart.order_service.event;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * Event published to Kafka topic "order.placed" after an order is saved.
 *
 * Consumed by: payment-service (picks it up and processes the payment)
 *
 * Message key: orderId (string) — ensures all events for the same order
 * go to the same Kafka partition, preserving ordering per order.
 */
public record OrderPlacedEvent(
        UUID orderId,
        UUID userId,
        String shopId,
        String productId,
        String productName,
        BigDecimal unitPrice,
        int quantity,
        BigDecimal totalAmount
) {}

File: localmart/payment-service/DTOs/OrderPlacedEvent.cs
namespace payment_service.DTOs;

// Matches exactly the Java record OrderPlacedEvent published by order-service to "order.placed" topic.
// Field names must match JSON keys in the Java record — C# and Java both serialize to camelCase by default.
//
// C# record = Java record. Immutable, auto-generates equality, ToString.
// { get; init; } = property that can only be set during object construction (immutable after).
public record OrderPlacedEvent(
    Guid OrderId,
    Guid UserId,
    string ShopId,
    string ProductId,
    string ProductName,
    decimal UnitPrice,
    int Quantity,
    decimal TotalAmount
);
